src/builder/commands/copy.rs

```diff
@@ -159,11 +185,33 @@ impl BuildStep for Copy {
     }
 }
 
-fn hash_path(hash: &mut Digest, path: &Path,
-    filter: &Filter, owner_uid: Option<uid_t>, owner_gid: Option<gid_t>)
+fn reset_permissions(path: &Path, umask: u32)
+    -> Result<(), StepError>
+{
+    let stat = try!(path.symlink_metadata()
```

Comment:
I'm not sure this is a good idea:

1. We just created a file, we should know what permission bits to apply
2. Now we depend on both `umask` from `Copy` command **and** the environment (i.e.what if umask in Copy is 002 and umask in environment is 020)